src/acc/generate_chunk.cpp

```diff
@@ -0,0 +1,167 @@
+    state_xmax_buffer[state] = globals.config.states[state].xmax;
+    state_ymin_buffer[state] = globals.config.states[state].ymin;
+    state_ymax_buffer[state] = globals.config.states[state].ymax;
+    state_radius_buffer[state] = globals.config.states[state].radius;
+    state_geometry_buffer[state] = globals.config.states[state].geometry;
+  }
+
+  // Kokkos::deep_copy (TO, FROM)
+
+  const int x_min = globals.chunk.tiles[tile].info.t_xmin;
+  const int x_max = globals.chunk.tiles[tile].info.t_xmax;
+  const int y_min = globals.chunk.tiles[tile].info.t_ymin;
+  const int y_max = globals.chunk.tiles[tile].info.t_ymax;
+
+  int xrange = (x_max + 2) - (x_min - 2) + 1;
+  int yrange = (y_max + 2) - (y_min - 2) + 1;
+
+  // Take a reference to the lowest structure, as Kokkos device cannot necessarily chase through the structure.
+
+  field_type &field = globals.chunk.tiles[tile].field;
+
+  const double state_energy_0 = state_energy_buffer[0];
+  const double state_density_0 = state_density_buffer[0];
+  const double state_xvel_0 = state_xvel_buffer[0];
+  const double state_yvel_0 = state_yvel_buffer[0];
+
+  const int base_stride = field.base_stride;
+  const int vels_wk_stride = field.vels_wk_stride;
+
+  // State 1 is always the background state
+  double *energy0 = field.energy0.data;
+  double *density0 = field.density0.data;
+  double *xvel0 = field.xvel0.data;
+  double *yvel0 = field.yvel0.data;
+
+#pragma acc parallel loop gang worker vector default(present) collapse(2) clover_use_target(globals.context.use_target)
+  for (int j = 0; j < (yrange); j++) {
+    for (int i = 0; i < (xrange); i++) {
+      energy0[i + j * base_stride] = state_energy_0;
+      density0[i + j * base_stride] = state_density_0;
+      xvel0[i + j * vels_wk_stride] = state_xvel_0;
+      yvel0[i + j * vels_wk_stride] = state_yvel_0;
+    }
+  }
+
+  for (int state = 1; state < globals.config.number_of_states; ++state) {
+
+    double *cellx = field.cellx.data;
+    double *celly = field.celly.data;
+
+    double *vertexx = field.vertexx.data;
+    double *vertexy = field.vertexy.data;
+
+    const double *state_density = state_density_buffer.data;
+    const double *state_energy = state_energy_buffer.data;
+    const double *state_xvel = state_xvel_buffer.data;
+    const double *state_yvel = state_yvel_buffer.data;
+    const double *state_xmin = state_xmin_buffer.data;
+    const double *state_xmax = state_xmax_buffer.data;
+    const double *state_ymin = state_ymin_buffer.data;
+    const double *state_ymax = state_ymax_buffer.data;
+    const double *state_radius = state_radius_buffer.data;
+    const int *state_geometry = state_geometry_buffer.data;
+
+/* TODO NVHPC says it can't parallelize the inner for loops
+ * We can try forcing if the index calcs are unique */
+#pragma acc parallel loop gang worker vector default(present) collapse(2) clover_use_target(globals.context.use_target)                            \
```

Comment:
In this particular case I'm not too worried because it's only used a few times for initialisation so we're just ensuring the data resides at the right place.
Please elaborate on what you mean by forcing it.